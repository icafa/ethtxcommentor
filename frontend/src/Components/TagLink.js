import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { shortenAddress, GU } from '../utils/utils'

const StyledLink = styled(Link)`
  display: inline-block;
  font-family: monospace;
  background: ${props =>
    props.background !== '' ? props.background : '#DDEAEF'};
  padding: ${GU / 2}px;
  border-radius: ${GU / 2}px;
  color: black;
  text-decoration: none;
  font-weight: normal;
`

const TagLink = ({ shorten, text, background = '', location }) => {
  let mutatedText = text
  if (shorten) {
    mutatedText = shortenAddress(text)
  }
  return (
    <StyledLink to={`${location}`} background={background}>
      {mutatedText}
    </StyledLink>
  )
}

TagLink.propTypes = {
  shorten: PropTypes.bool,
  text: PropTypes.string.isRequired,
  background: PropTypes.string,
  location: PropTypes.string.isRequired,
}

export default TagLink
