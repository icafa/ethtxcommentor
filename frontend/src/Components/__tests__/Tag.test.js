import React from 'react'
import { render } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import TagLink from '../TagLink'

describe('Tag component', () => {
  it('renders correctly', () => {
    const history = createMemoryHistory()
    const { asFragment } = render(
      <Router history={history}>
        <TagLink location="/" text="hello" />
      </Router>
    )
    expect(asFragment).toMatchSnapshot()
  })
})
