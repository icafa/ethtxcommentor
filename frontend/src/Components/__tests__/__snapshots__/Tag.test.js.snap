// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component renders correctly 1`] = `[Function]`;
